contracts/utils/ItsHubAddressTracker.sol

```diff
@@ -34,8 +38,10 @@ abstract contract ItsHubAddressTracker is IItsHubAddressTracker {
         bytes32 itsHubAddressMiddle_ = itsHubAddressMiddle;
         uint8 itsHubAddressSuffix_ = itsHubAddressSuffix;
         hubAddress = new string(65);
-        assembly {
+
+        assembly ('memory-safe') {
             mstore(add(hubAddress, 32), itsHubAddressPrefix_)
+            // The below writes 32 bytes into the middle slot and 1 bytes into the suffix slot, so it is done before the middle write.
```

Comment:
```suggestion
            // This writes the 1-byte suffix to the 65th position, but also 31 other bytes into the middle. Hence, the middle is written after this to prevent it from being overwritten.
```